From 9da1996267eed65897a74d9bdb39e4401b156e7d Mon Sep 17 00:00:00 2001
From: Ben Walton <bwalton@opencsw.org>
Date: Thu, 24 Mar 2011 15:50:11 +0100
Subject: [PATCH 1/2] Use LOG_PID instead of LOG_PERROR in Syslog.open test

LOG_PERROR isn't a POSIX option for syslog, so it fails on platforms
that don't define it.  Solaris 9 and 10 are examples of this.

Use LOG_PID instead.

Signed-off-by: Ben Walton <bwalton@opencsw.org>
---
 test/test_syslog.rb |    5 +++--
 1 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/test/test_syslog.rb b/test/test_syslog.rb
index c508412..265b0d7 100644
--- a/test/test_syslog.rb
+++ b/test/test_syslog.rb
@@ -43,10 +43,11 @@ class TestSyslog < Test::Unit::TestCase
     Syslog.close
 
     # given parameters
-    Syslog.open("foo", Syslog::LOG_NDELAY | Syslog::LOG_PERROR, Syslog::LOG_DAEMON)
+    options = Syslog::LOG_NDELAY | Syslog::LOG_PID
+    Syslog.open("foo", options, Syslog::LOG_DAEMON)
 
     assert_equal('foo', Syslog.ident)
-    assert_equal(Syslog::LOG_NDELAY | Syslog::LOG_PERROR, Syslog.options)
+    assert_equal(options, Syslog.options)
     assert_equal(Syslog::LOG_DAEMON, Syslog.facility)
 
     Syslog.close
-- 
1.7.3.2

